torch/_dynamo/variables/torch.py

```diff
@@ -365,6 +373,26 @@ def get_state_from_generator():
                 ),
                 **options,
             )
+        if (
+            all([isinstance(x, DynamicShapeVariable) for x in args])
+        ):
+            if self.value == math.sqrt:
```

Comment:
jansel: The variable builder or constructor for math.sqrt should have replaced it with sym_sqrt, instead of doing it at the call site. Can have a dictionary of substitutions mapping the original function to the replacement function

Reply:
Counterpoint - you do not know if you need sym_sqrt until the args come in, unless we want to always replace it, which I am not sure we do?

Reply:
One piece of prior data we have is that we want to replace `int(a)` from user code with `sym_int(a)`, and we always do it - if `a` happens to not be symbolic, `sym_int()` knows how to handle it. We could treat `sym_sqrt` similarly?